File: java-shot-api/JavaShotAPI/JavaShotAPI/Models/Pregunta.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace JavaShotAPI.Models
{
    [Table("Pregunta")]
    public class Pregunta
    {
        public Guid PreguntaID { get; set; }
        public string Titulo { get; set; }
        public DateTime FechaRegistro { get; set; }
        public string ExplicacionRespuesta { get; set; }

        public Pregunta(string titulo, string explicacionRespuesta)
        {
            PreguntaID = Guid.NewGuid();
            Titulo = titulo;
            FechaRegistro = DateTime.Now;
            ExplicacionRespuesta = explicacionRespuesta;
        }
    }
}

File: java-shot-web/src/app/models/pregunta.ts
import { Respuesta } from './respuesta';

export class Pregunta {
    constructor(
        public preguntaID?: string,
        public titulo?: string,
        public fechaRegistro?: Date,
        public contestada?: boolean,
        public contestadaCorrectamente?: boolean,
        public partidaPreguntaID?: string,
        public respuestas?: Respuesta[],
        public explicacionRespuesta?: string
    ) {}
}
